CreatePr.cmd

```diff
@@ -117,13 +117,31 @@ REM ************************************************************
 			goto :Error
 		)
 
+	:ExtractRemote
+		//for /F "usebackq" %%i in (`git remote -v ^| findstr /VC:"Microsoft/BuildXL.git"`) do set PR_REMOTE=%%i
+		REM: the Delims are tab and space intentionally
+		for /F "usebackq tokens=1,2 delims=	 " %%i in (`git remote -v ^| findstr /VC:"Microsoft/BuildXL.git"`) do (
+			set PR_REMOTE_NAME=%%i
+			set PR_REMOTE_URI=%%j
+		)
+		if "%PR_REMOTE_NAME%" == "" (
+			echo --------------------------------------------------------------------------------------
+			echo - FAILURE  :-(  Could not determine a git 'remote' that is not Microsoft/BuildXL.git -
+			echo --------------------------------------------------------------------------------------
+			exit /b 1
+			goto :Error
+		)
+
 	:ForcePushToServer
-		call git push --force origin HEAD:%PR_BRANCH_NAME%
+		call git push --force %PR_REMOTE_NAME% HEAD:%PR_BRANCH_NAME%

```

Comment:
Why push to "upstream" and not "origin"?  I don't think we should be doing this.  This `CreatePr.cmd` script is available for external user and they will not have permissions to do this.  But even we should follow best practices and creating branches in our own forks instead of in Microsoft/BuildXL.git

Reply:
Because I don't want to be a branch manager and FI/RI I clone master from the original repo, work on that and then push to my remote. So this code is now generic regarding if you cloned the main repo or if you cloned your fork, this will always push to your fork and there doesn't have to be a convention on how you name your remotes.

Reply:
I just realized that, unsurprisingly, I didn't interpret this batch script correctly, so in fact `PR_REMOTE_NAME` indeed refers to the name of the remote that is not Microsoft/BuildXL.git, which is great.